pkg/query-service/app/traces/v3/query_builder.go

```diff
@@ -404,17 +425,19 @@ func addOffsetToQuery(query string, offset uint64) string {
 }
 
 func PrepareTracesQuery(start, end int64, queryType v3.QueryType, panelType v3.PanelType, mq *v3.BuilderQuery, keys map[string]v3.AttributeKey) (string, error) {
-	query, err := buildTracesQuery(start, end, mq.StepInterval, mq, constants.SIGNOZ_SPAN_INDEX_TABLENAME, keys)
+	query, err := buildTracesQuery(start, end, mq.StepInterval, mq, constants.SIGNOZ_SPAN_INDEX_TABLENAME, keys, panelType)
 	if err != nil {
 		return "", err
 	}
 	if panelType == v3.PanelTypeValue {
 		query, err = reduceToQuery(query, mq.ReduceTo, mq.AggregateOperator)
 	}
-	query = addLimitToQuery(query, mq.Limit, panelType)
+	if panelType != v3.PanelTypeTrace {
```

Comment:
The condition here is `if panelType != v3.PanelTypeTrace ` but there is a change in `addLimitToQuery` with  `if ... || panelType == v3.PanelTypeTrace` which is the opposite. I understand what is happening here, but it will be hard for future readers to make sense of what is happening. I would consider refactoring.